### DISCUSSION TITLE
[Request] Headscale Controller (Self-hosted Tailscale Contoller Server)

### DISCUSSION THREAD

**Discussion by yodaphone** [tteck/Proxmox#1137] [Ideas]
Headscale - an open source self hosted Tailscale Controller that works with Tailscale clients

https://github.com/juanfont/headscale

There's an UI as well for it

https://github.com/ifargle/headscale-webui

**Comment by 414YogaDog** [comment]
I would love to see this.

**Comment by tteck** [comment]
Done

```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/headscale.sh)"
```
